widget/embedded/src/store/slices/data.test.ts

```diff
@@ -351,3 +356,89 @@ describe('search in tokens', () => {
     expect(secondResult).toBe('RNG');
   });
 });
+
+describe('supported tokens from config', () => {
```

Comment:
We need a test to cover cache is hitting or not. Something like this:

```js
test('Check tokesn calculation is caching', () => {
    const rangoToken = customTokens[0];
    const djangoToken = customTokens[1];

    appStoreState = updateAppStoreConfig(appStoreState, {
      from: {
        tokens: [rangoToken, djangoToken],
      },
    });

    expect(cacheService.get('supportedSourceTokens')?.length || 0).toBe(0);
    appStoreState.tokens({
      type: 'source',
    });

    expect(cacheService.get('supportedSourceTokens')?.length || 0).toBe(1);
  });
```

Reply:
The whole point of this is adding a cache layer. we should cover its behavior using tests. like previous comment. You can also check updaing, deleting and other cases if you have.